<!DOCTYPE html>
<html>
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<title>Markmap</title>
<style>
* {
  margin: 0;
  padding: 0;
}
#mindmap {
  display: block;
  width: 100vw;
  height: 100vh;
}
</style>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/markmap-toolbar@0.17.0/dist/style.css">
</head>
<body>
<svg id="mindmap"></svg>
<script src="https://cdn.jsdelivr.net/npm/d3@7.8.5/dist/d3.min.js"></script><script src="https://cdn.jsdelivr.net/npm/markmap-view@0.17.0/dist/browser/index.js"></script><script src="https://cdn.jsdelivr.net/npm/markmap-toolbar@0.17.0/dist/index.js"></script><script>(()=>{setTimeout(()=>{const{markmap:O,mm:h}=window,M=new O.Toolbar;M.attach(h);const oe=M.render();oe.setAttribute("style","position:absolute;bottom:20px;right:20px"),document.body.append(oe)})})()</script><script>((i,L,f,r)=>{const w=i();window.mm=w.Markmap.create("svg#mindmap",(L||w.deriveOptions)(r),f)})(()=>window.markmap,null,{"content":"Excepciones en Python","children":[{"content":"Definición y Tipos de Errores","children":[{"content":"\n<p data-lines=\"5,6\">En Python, las excepciones son errores detectados durante la ejecución del programa, a diferencia de los errores sintácticos que se detectan en el tiempo de compilación. Hay dos tipos principales de errores:</p>","children":[],"payload":{"lines":"5,7"}},{"content":"\n<p data-lines=\"7,8\"><strong>Errores Sintácticos:</strong> Estos ocurren cuando el código no sigue la sintaxis correcta de Python.</p>","children":[],"payload":{"lines":"7,8"}},{"content":"\n<p data-lines=\"8,9\"><strong>Excepciones:</strong> Estos son errores detectados durante la ejecución que interrumpen el flujo normal del programa.</p>","children":[],"payload":{"lines":"8,10"}}],"payload":{"lines":"3,4"}},{"content":"Ejemplos de Excepciones Comunes","children":[{"content":"<strong>ZeroDivisionError:</strong> Ocurre al intentar dividir cualquier número por cero.","children":[],"payload":{"lines":"12,13"}},{"content":"<strong>NameError:</strong> Se produce cuando una variable no está definida.","children":[],"payload":{"lines":"13,14"}},{"content":"<strong>TypeError:</strong> Resulta de una operación o función aplicada a un objeto de tipo inapropiado.","children":[],"payload":{"lines":"14,16"}}],"payload":{"lines":"10,11"}},{"content":"Manejo de Excepciones","children":[{"content":"\n<p data-lines=\"17,18\">El manejo de excepciones en Python se realiza a través de los bloques <code>try</code> y <code>except</code>. Esto permite que el programa continúe incluso después de que ocurra un error.</p>","children":[{"content":"\n<pre><code class=\"language-python\">  <span class=\"hljs-keyword\">try</span>:\n      <span class=\"hljs-comment\"># Código que puede lanzar una excepción</span>\n      x = <span class=\"hljs-number\">1</span> / <span class=\"hljs-number\">0</span>\n  <span class=\"hljs-keyword\">except</span> ZeroDivisionError:\n      <span class=\"hljs-comment\"># Código que se ejecuta si hay una excepción</span>\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"No se puede dividir por cero.\"</span>)\n</code></pre>","children":[],"payload":{"lines":"19,28"}}],"payload":{"lines":"17,28"}}],"payload":{"lines":"16,17"}},{"content":"Especificando Excepciones","children":[{"content":"\n<p data-lines=\"29,30\">Python permite especificar múltiples tipos de excepciones en el bloque <code>except</code> para manejar diferentes errores de manera diferente.</p>","children":[{"content":"\n<pre><code class=\"language-python\">  <span class=\"hljs-keyword\">try</span>:\n      x = <span class=\"hljs-built_in\">int</span>(<span class=\"hljs-built_in\">input</span>(<span class=\"hljs-string\">\"Introduce un número entero: \"</span>))\n  <span class=\"hljs-keyword\">except</span> ValueError:\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"Eso no es un entero.\"</span>)\n  <span class=\"hljs-keyword\">except</span> (TypeError, NameError):\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"Error de tipo o NameError detectado.\"</span>)\n</code></pre>","children":[],"payload":{"lines":"31,40"}}],"payload":{"lines":"29,40"}}],"payload":{"lines":"28,29"}},{"content":"Cláusulas <code>else</code> y <code>finally</code>","children":[{"content":"\n<p data-lines=\"42,43\"><strong>Else:</strong> Se ejecuta si el bloque <code>try</code> no lanza una excepción.</p>","children":[],"payload":{"lines":"42,43"}},{"content":"\n<p data-lines=\"43,44\"><strong>Finally:</strong> Se ejecuta después de los bloques <code>try</code> y <code>except</code>, independientemente de si se lanzó una excepción o no.</p>","children":[{"content":"\n<pre><code class=\"language-python\">  <span class=\"hljs-keyword\">try</span>:\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"Intentamos ejecutar el bloque try:\"</span>)\n      x = <span class=\"hljs-number\">1</span> / <span class=\"hljs-number\">1</span>\n  <span class=\"hljs-keyword\">except</span>:\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"Algo salió mal...\"</span>)\n  <span class=\"hljs-keyword\">else</span>:\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"¡Todo bien!\"</span>)\n  <span class=\"hljs-keyword\">finally</span>:\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">\"El bloque 'finally' siempre se ejecuta.\"</span>)\n</code></pre>","children":[],"payload":{"lines":"45,57"}}],"payload":{"lines":"43,57"}}],"payload":{"lines":"40,41"}},{"content":"Lanzamiento de Excepciones","children":[{"content":"\n<p data-lines=\"59,60\">Se pueden lanzar excepciones intencionadamente usando la palabra clave <code>raise</code>. Esto es útil para crear verificaciones y hacer que el programa falle deliberadamente bajo condiciones específicas.</p>","children":[{"content":"\n<pre><code class=\"language-python\">   x = <span class=\"hljs-number\">10</span>\n   <span class=\"hljs-keyword\">if</span> x &gt; <span class=\"hljs-number\">5</span>:\n       <span class=\"hljs-keyword\">raise</span> Exception(<span class=\"hljs-string\">\"x no debe exceder de 5. El valor de x fue: {}\"</span>.<span class=\"hljs-built_in\">format</span>(x))\n</code></pre>","children":[],"payload":{"lines":"61,67"}}],"payload":{"lines":"59,67"}}],"payload":{"lines":"57,58"}},{"content":"Definición de Excepciones Personalizadas","children":[{"content":"\n<p data-lines=\"68,69\">Se pueden crear excepciones personalizadas mediante la definición de clases que se derivan de la clase Exception. Esto permite crear excepciones que son específicas para las necesidades del programa.</p>","children":[{"content":"\n<pre><code class=\"language-python\">  <span class=\"hljs-keyword\">class</span> <span class=\"hljs-title class_\">MiError</span>(<span class=\"hljs-title class_ inherited__\">Exception</span>):\n      <span class=\"hljs-keyword\">def</span> <span class=\"hljs-title function_\">__init__</span>(<span class=\"hljs-params\">self, valor</span>):\n          self.valor = valor\n      <span class=\"hljs-keyword\">def</span> <span class=\"hljs-title function_\">__str__</span>(<span class=\"hljs-params\">self</span>):\n          <span class=\"hljs-keyword\">return</span> <span class=\"hljs-built_in\">repr</span>(self.valor)\n\n  <span class=\"hljs-keyword\">try</span>:\n      <span class=\"hljs-keyword\">raise</span> MiError(<span class=\"hljs-number\">2</span>*<span class=\"hljs-number\">2</span>)\n  <span class=\"hljs-keyword\">except</span> MiError <span class=\"hljs-keyword\">as</span> e:\n      <span class=\"hljs-built_in\">print</span>(<span class=\"hljs-string\">'Mi excepción ocurrió, valor:'</span>, e.valor)\n</code></pre>","children":[],"payload":{"lines":"70,82"}}],"payload":{"lines":"68,82"}}],"payload":{"lines":"67,68"}}],"payload":{"lines":"1,2"}},{"colorFreezeLevel":3,"maxWidth":600})</script>
</body>
</html>
